pilight throttle test fails intermittently

## The problem
<!-- 
  Describe the issue you are experiencing here to communicate to the
  maintainers. Tell us what you were trying to do and what happened.
-->
`FAILED tests/components/pilight/test_init.py::TestPilightCallrateThrottler::test_call_rate_delay_throttle_enabled`
```
2020-08-31T17:00:15.6631723Z ______ TestPilightCallrateThrottler.test_call_rate_delay_throttle_enabled ______
2020-08-31T17:00:15.6632484Z [gw0] linux -- Python 3.7.1 /__w/core/core/venv/bin/python
2020-08-31T17:00:15.6632564Z 
2020-08-31T17:00:15.6632716Z self = <tests.components.pilight.test_init.TestPilightCallrateThrottler testMethod=test_call_rate_delay_throttle_enabled>
2020-08-31T17:00:15.6632831Z 
2020-08-31T17:00:15.6632971Z     def test_call_rate_delay_throttle_enabled(self):
2020-08-31T17:00:15.6633117Z         """Test that throttling actually work."""
2020-08-31T17:00:15.6633272Z         runs = []
2020-08-31T17:00:15.6633387Z         delay = 5.0
2020-08-31T17:00:15.6633515Z     
2020-08-31T17:00:15.6633651Z         limit = pilight.CallRateDelayThrottle(self.hass, delay)
2020-08-31T17:00:15.6633800Z         action = limit.limited(lambda x: runs.append(x))
2020-08-31T17:00:15.6634096Z     
2020-08-31T17:00:15.6634216Z         for i in range(3):
2020-08-31T17:00:15.6634674Z             action(i)
2020-08-31T17:00:15.6634781Z     
2020-08-31T17:00:15.6635042Z >       assert runs == []
2020-08-31T17:00:15.6635152Z E       AssertionError: assert [0] == []
2020-08-31T17:00:15.6635261Z E         Left contains one more item: 0
2020-08-31T17:00:15.6635365Z E         Full diff:
2020-08-31T17:00:15.6635607Z E         - []
2020-08-31T17:00:15.6635713Z E         + [0]
2020-08-31T17:00:15.6635816Z E         ?  +
2020-08-31T17:00:15.6635867Z 
2020-08-31T17:00:15.6635978Z tests/components/pilight/test_init.py:406: AssertionError
```

--- a/tests/components/pilight/test_init.py
+++ b/tests/components/pilight/test_init.py
@@ -403,7 +403,8 @@ class TestPilightCallrateThrottler(unittest.TestCase):
         for i in range(3):
             action(i)
 
-        assert runs == []
+        self.hass.block_till_done()
+        assert runs == [0]
 
         exp = []
         now = dt_util.utcnow()

